### PR TITLE
truck inline block

### LINKED ISSUE #16: Trucks lineup carousel
Related to #36 

Restyling of the block [tabbed carousel](https://www.volvotrucks.us/trucks/vnr-electric/#meet-the-vnr-family). It should be a fork/new block, as we’re not using the same structure for every occurance of this block. The code of this block will also be used for Mack Trucks.

- First truck enter when component comes in view
- Clicking a tab would navigate in the trucks accordingly
- Different images are used on Tablet/Mobile (trucks are front facing)

When the tabs won’t fit, we can scroll through them (overflow: auto)

### Content
The section should start with an H2 (now displayed as H4 or H3)

Every slide/truck will have its own content:
- Product name (displayed inline with text)
- Text
- Explore button links to its respective PDP
- Build yours button links to an external URL and can be shown optionally

On the PDP, the section and block will be displayed differently:
- Text below heading
- Different button type, which links to Truck Builder
- Link with + icon (to a modal window?)

### Design
The [item on Figma](https://www.figma.com/file/IcqreRKnNYyDVMc28yoOeh/Homepage?type=design&node-id=1387-17455&mode=dev) is shown on the homepage.

On the PDP, we show instead of the different truck series, the models within the series.

#### Out of scope
- Animation of moving truck on scroll
- Animation of rolling tyres

### PR DESCRIPTION
**Pull request by kesiah**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #16

Test URLs:
- Before: https://main--vg-volvotrucks-us-rd--netcentric.hlx.page/drafts/laura/
- After: https://16-trucks-lineup--vg-volvotrucks-us-rd--netcentric.hlx.page/drafts/laura/

### CODE REVIEW

File: blocks/v2-tabbed-carousel/v2-tabbed-carousel.css [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+}
+
+/* Navigation */
+ul.v2-tabbed-carousel__navigation {
+  display: flex;
+  flex-flow: row nowrap;
+  overflow: auto;
+  list-style: none;
+  margin: 35px 0;
+  padding: 0 35px;
```
Comment (cogniSyb) [inline blocks/v2-tabbed-carousel/v2-tabbed-carousel.css:146]: Changing this to `padding: 2px 35px` would make the button focus state properly visible

File: blocks/v2-tabbed-carousel/v2-tabbed-carousel.css [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+  margin: 10px 0;
+}
+
+/* Navigation */
+ul.v2-tabbed-carousel__navigation {
+  display: flex;
+  flex-flow: row nowrap;
+  overflow: auto;
+  list-style: none;
+  margin: 35px 0;
```
Comment (cogniSyb) [inline blocks/v2-tabbed-carousel/v2-tabbed-carousel.css:145]: Could we use here (and on following line) 32px instead?

File: blocks/v2-tabbed-carousel/v2-tabbed-carousel.css [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+}
+
+.v2-tabbed-carousel__navigation::before,
+.v2-tabbed-carousel__navigation::after {
+  content: '';
+  margin: auto;
+}
+
+.v2-tabbed-carousel__navigation-item {
+  margin: 0 15px;
```
Comment (cogniSyb) [inline blocks/v2-tabbed-carousel/v2-tabbed-carousel.css:169]: Could we use 16px here instead?

File: blocks/v2-tabbed-carousel/v2-tabbed-carousel.css [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+
+.v2-tabbed-carousel__desc-item.active {
+  opacity: 1;
+  transform: translateX(0);
+  transition: opacity var(--duration-large) var(--easing-entrance),
+    transform 1s var(--easing-entrance);
+}
+
+.v2-tabbed-carousel__text {
+  text-wrap: balance;
```
Comment (cogniSyb) [inline blocks/v2-tabbed-carousel/v2-tabbed-carousel.css:115]: Should get a max-width. Around 400px?

**Review by cogniSyb — CHANGES_REQUESTED** [review]

**Review by cogniSyb — APPROVED** [review]
I’ve opened issue #36 to do the finishing touch on some visual details of this component. For now this PR is approved :)

File: blocks/v2-tabbed-carousel/v2-tabbed-carousel.js [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+
+function buildTabNavigation(tabItems, clickHandler) {
+  const tabNavigation = createElement('ul', `${blockName}__navigation`);
+  const navigationLine = createElement('li', `${blockName}__navigation-line`);
+  let timeout;
+
+  [...tabItems].forEach((tabItem, i) => {
+    const listItem = createElement('li', `${blockName}__navigation-item`);
+    const button = createElement('button');
+    button.addEventListener('click', () => clickHandler.call(this, i));
```
Comment (TomaszDziezykNetcentric) [inline blocks/v2-tabbed-carousel/v2-tabbed-carousel.js:21]: `this` for the arrow function is undefined. Why not just call the `clickHandler(tabItems, i)`?

File: blocks/v2-tabbed-carousel/v2-tabbed-carousel.js [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+    if (navigationLine) {
+      button.addEventListener('mouseover', (e) => {
+        clearTimeout(timeout);
+        Object.assign(navigationLine.style, {
+          left: `${e.currentTarget.getBoundingClientRect().x + tabNavigation.scrollLeft}px`,
```
Comment (TomaszDziezykNetcentric) [inline blocks/v2-tabbed-carousel/v2-tabbed-carousel.js:26]: A small simplification:
```
const {x, scrollLeft, width} = e.currentTarget.getBoundingClientRect();
Object.assign(navigationLine.style, {
    left: `${x + scrollLeft}px`,
    width:  `${width}px`
}
```

File: blocks/v2-tabbed-carousel/v2-tabbed-carousel.js [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+          width: `${e.currentTarget.getBoundingClientRect().width}px`,
+        });
+      });
+
+      button.addEventListener('mouseout', () => {
+        timeout = setTimeout(() => {
+          const activeItem = document.querySelector(`.${blockName}__navigation-item.active`);
+          Object.assign(navigationLine.style, {
+            left: `${activeItem.getBoundingClientRect().x + tabNavigation.scrollLeft}px`,
```
Comment (TomaszDziezykNetcentric) [inline blocks/v2-tabbed-carousel/v2-tabbed-carousel.js:35]: Simplification as above:
```
const {x, scrollLeft, width} = activeItem.getBoundingClientRect();
Object.assign(navigationLine.style, {
    left: `${x + scrollLeft}px`,
    width:  `${width}px`
}
```

File: blocks/v2-tabbed-carousel/v2-tabbed-carousel.js [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+
+  [images, descriptions, navigation].forEach((c) => c.querySelectorAll('.active').forEach((i) => i.classList.remove('active')));
+  images.children[index].classList.add('active');
+  descriptions.children[index].classList.add('active');
+  navigation.children[index].classList.add('active');
+
+  if (navigationLine) {
+    const activeNavigationItem = navigation.children[index];
+    Object.assign(navigationLine.style, {
+      left: `${activeNavigationItem.getBoundingClientRect().x + navigation.scrollLeft}px`,
```
Comment (TomaszDziezykNetcentric) [inline blocks/v2-tabbed-carousel/v2-tabbed-carousel.js:67]: As above code simplification:
```
const {x, scrollLeft, widht} = activeNavigationItem.getBoundingClientRect();
```

File: blocks/v2-tabbed-carousel/v2-tabbed-carousel.js [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+      parentTextContainer.appendChild(textContainer);
+    }
+
+    // Wrap links in container
+    const buttonContainer = createElement('div', `${blockName}__buttons-container`);
+    const buttons = tabContent.querySelectorAll('.button-container');
+
+    buttons.forEach((bt, i) => {
+      const buttonLink = bt.firstElementChild;
+      // buttonLink.classList.remove('primary');
```
Comment (TomaszDziezykNetcentric) [inline blocks/v2-tabbed-carousel/v2-tabbed-carousel.js:215]: Please remove commented code

**Review by TomaszDziezykNetcentric — CHANGES_REQUESTED** [review]

File: blocks/v2-tabbed-carousel/v2-tabbed-carousel.css [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+.v2-tabbed-carousel__navigation button {
+  background: 0 0;
+  border: none;
+  color: var(--text-color);
+  font-size: var(--f-button-font-size);
+  letter-spacing: var(--f-button-letter-spacing);
+  line-height: 100%;
+  margin: 0;
+  padding: 12px 0;
+  text-transform: uppercase;
```
[the remark this answers is not in the document]
Comment (kesiah) [reply blocks/v2-tabbed-carousel/v2-tabbed-carousel.css:183]: In this case, I'm not sure, we decided on this approach for the buttons, but also for the navigation bar items?
wdyt @cogniSyb ?

[the remark this answers is not in the document]
Comment (cogniSyb) [reply blocks/v2-tabbed-carousel/v2-tabbed-carousel.css:182]: In this case it’s a design decision that the buttons will always be capitalised. At least for this component. If this block would be created as reusable (tabbed) component, then I would’ve opted indeed for letting the content editor decide.

**Review by TomaszDziezykNetcentric — APPROVED** [review]

**Pull request merged**